### PR TITLE
Summary completion

### PR DESCRIPTION
**Pull request by joakim-hove**
WGITL, WGIRL, WWITL, WWIRL, WGPTL, WGPRL, WLPTL, WOPTL, WOPRL, WWPTL and WWPRL

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp:443]: Doesn't `fmt::format()`  already do the requisite padding/filling?
```C++
#include <fmt/format.h>

int main()
{
    fmt::print("{}{:_>3}\n", "COPRL",   5);
    fmt::print("{}{:_>3}\n", "COPRL",  15);
    fmt::print("{}{:_>3}\n", "COPRL", 123);
}
// Output:
// COPRL__5
// COPRL_15
// COPRL123
```

Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp:443]: Cool - I did not know; will update.

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp:1124]: You might consider using `std::string::pop_back()` instead of these `.substr()` calls:
```C++
if (is_well_completion(keyword) || is_connection_completion(keyword))
    keyword.pop_back();  // Remove trailing 'L' character
```

File: src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp:461]: This is more expensive than it needs to be because you end up parsing out the same information for each node.  If you use the `SummaryConfigNode` class like the other processing functions then you only need to change the information that actually differs between each node (i.e., the `.number()`).

**Review by bska — APPROVED** [review]
This looks good to me overall.  I noticed a couple of potential improvements which you can address at your own volition.

File: src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp
Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp:1124]: Good idea - will do it.

File: src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp
Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp:461]: Well - it is not exactly expensive - in the great scheme of things. The point was that the current approach allowed a more local/compact approach - which I liked. I'll take a look though.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by alfbr** [comment]
Just give me a heads-up when this is ready for testing.

**Comment by joakim-hove** [comment]
@alfbr : This is ready for testing

**Comment by joakim-hove** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Looks good to me.  I'll defer to the build check and upstream testing to give final merge approval.

**Comment by alfbr** [comment]
There are a couple of discrepancies, so please hold off merging until we have investigated.

**Comment by joakim-hove** [comment]
OK  - this has been verified with Equinor; one last build and then I will merge.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**
